[JohannsenLum] ip

JohannsenLum opened the pull request:
# ErelBot

> "A task is not done until it is done" - Louis Fried ([source](https://plaky.com/blog/project-management-quotes/))

Erelbot is your ultimate task management solution 👍 . It's 

- **Text-based** for simplicity
- *Easy to learn* and use

## Features:
- [x] Task management
- [ ] Deadline tracking (coming soon)
- [ ] start reminders (coming soon)


### How to get started:
1. Download Erelbot.jar from [here](https://github.com/JohannsenLum/ip/releases).
2. Double-click the downloaded file.
3. Start adding your tasks!
4. Let Erelbot organize your life 😎

If you are a Java enthusiast, Here's the `main` method: 
```
public static void main(String[] args) {
        new Erel("./data/erel.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Renaming, added greetings, exit

Commit:
Echoing User's commands

Commit:
Updated echo with correct spaces

Commit:
Added add, list

Commit:
Mark as Done, added new class Task

Commit:
Created Todos, Events, Deadlines

Commit:
Automated Text UI Testing successfully set up

Commit:
Added Error Handling

Commit:
Added Delete function and changed if-else statements to switch cases

Commit:
Added Enums

Commit:
Reformat greet and exit

Commit:
Add all Exceptions

Commit:
Refactor TODO case

Commit:
Refactor DEADLINE case

Commit:
Refactor EVENT case

Commit:
Update Automated Text UI Testing

Commit:
Add Javadocs

Commit:
Update Java Docs

Commit:
Remove wrong docs

Commit:
Implement Level-7: Save task list to file

Commit:
Merge branch 'branch-Level-7'

This merge integrates the Level-7 feature, which adds the ability to save and load tasks from a file

Commit:
Implement Level-8: Add date/time support for deadlines

Commit:
Merge branch 'branch-Level-8'
This merge integrates the Level-8 feature, which adds date/time support for deadlines and events

Commit:
Refactor code for A-MoreOOP: Extract Ui, Storage, Parser, TaskList and Action classes

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Update text-ui-test inputs and expected

Commit:
Implement A-Packages:Reorganized classes into packages

Commit:
Merge branch 'A-Packages'
Implement A-Packages: Organize classes into suitable java packages.

Commit:
Merge branch 'add-gradle-support'

Commit:
Implement A-Gradle: Use Gradle

Commit:
Implement A-JUnit: Add JUnit Tests

Commit:
Merge branch 'branch-A-JUnit'

Commit:
Reformat Erel.java and Parser.java

Commit:
Add JavaDocs

Commit:
Reformat Erel.java to follow coding standards

Commit:
Implement Level-9: Find action

Commit:
Merge branch 'branch-A-CodingStandard'
Reformat code to follow Java Coding Standards"

Commit:
Merge branch 'branch-Level-9'
Implement Find action to allow users to find tasks using keywords

Commit:
Solve merge-conflicts

Commit:
Add CheckStyle configuration

Commit:
Add basic GUI for Erel

xplus2g4 commented on src/main/java/erel/command/Action.java:
```diff
@@ -0,0 +1,21 @@
+ * Represents an action that can be executed in the Erel bot. Actions are commands that perform specific operations on
+ * the task list, such as adding, deleting, or marking tasks as done.
+ */
+public interface Action {
```
If the interface is providing a capability, try naming it "-able". i.e. Actionable

xplus2g4 commented on src/main/java/erel/command/TodoAction.java:
```diff
@@ -0,0 +1,36 @@
+ * description and adds it to the task list, and saves the updated list to storage.
+ */
+public class TodoAction implements Action {
+    String description;
```
`private` modifier preferred

xplus2g4 commented on src/main/java/erel/parser/Parser.java:
```diff
@@ -0,0 +1,194 @@
+     * @throws DateTimeParseException If the input string does not match the expected format.
+     */
+    public static LocalDateTime parseDateTime(String input) throws DateTimeParseException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```
Avoid magic strings, make it a constant.

xplus2g4 commented on src/main/java/erel/command/DeleteAction.java:
```diff
@@ -0,0 +1,35 @@
+     * @throws Exception If an error occurs during the execution of the action (e.g., invalid task number).
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws Exception {
```
Avoid throwing generic checked exceptions, might be confusing to handle

xplus2g4 commented on src/main/java/erel/command/Action.java:
```diff
@@ -0,0 +1,21 @@
+     * @param storage The storage for saving or loading tasks.
+     * @throws Exception If an error occurs during the execution of the action.
+     */
+    void execute(TaskList tasks, Ui ui, Storage storage) throws Exception;
```
Avoid throwing generic checked exceptions, might be confusing to handle

xplus2g4 commented on src/main/java/erel/command/DeadlineAction.java:
```diff
@@ -0,0 +1,38 @@
+     * @param storage The storage for saving the updated task list.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```
Will it throw exceptions?

xplus2g4 commented on src/main/java/erel/command/EventAction.java:
```diff
@@ -0,0 +1,41 @@
+     * @throws Exception If an error occurs during the execution of the action.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws Exception {
```
Avoid throwing generic checked exceptions, might be confusing to handle

xplus2g4 commented on src/main/java/erel/command/ExitAction.java:
```diff
@@ -0,0 +1,26 @@
+     * @throws Exception If an error occurs during the execution of the action.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws Exception {
```
Avoid throwing generic checked exceptions, might be confusing to handle

xplus2g4 commented on src/main/java/erel/command/FindAction.java:
```diff
@@ -0,0 +1,34 @@
+     * @throws Exception If an error occurs during the execution of the action.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws Exception {
```
Avoid throwing generic checked exceptions, might be confusing to handle

xplus2g4 commented on src/main/java/erel/command/MarkAction.java:
```diff
@@ -0,0 +1,37 @@
+     * @throws Exception If an error occurs during the execution of the action.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws Exception {
```
Avoid throwing generic checked exceptions, might be confusing to handle

xplus2g4 commented on src/main/java/erel/command/PrintListAction.java:
```diff
@@ -0,0 +1,25 @@
+     * @throws Exception If an error occurs during the execution of the action.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws Exception {
```
Avoid throwing generic checked exceptions, might be confusing to handle

xplus2g4 commented on src/main/java/erel/command/TodoAction.java:
```diff
@@ -0,0 +1,36 @@
+     * @throws Exception If an error occurs during the execution of the action.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws Exception {
```
Avoid throwing generic checked exceptions, might be confusing to handle

xplus2g4 commented on src/main/java/erel/command/UnmarkAction.java:
```diff
@@ -0,0 +1,39 @@
+     * @throws Exception If an error occurs during the execution of the action.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws Exception {
```
Avoid throwing generic checked exceptions, might be confusing to handle

xplus2g4 commented on src/main/java/erel/storage/Storage.java:
```diff
@@ -0,0 +1,74 @@
+                writer.write(task.toFileFormat() + System.lineSeparator());
+            }
+        } catch (IOException e) {
+            System.out.println("Error saving tasks to file" + e.getMessage());
```
maybe it'll be better to throw the IOException directly?

xplus2g4 commented on src/main/java/erel/task/Deadline.java:
```diff
@@ -0,0 +1,66 @@
+        String[] lines = fileFormat.split(" \\| ");
+        boolean isDone = lines[1].equals("1");
+        String descrption = lines[2];
+        LocalDateTime by = LocalDateTime.parse(lines[3], DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
```
Avoid magic string, use a constant

xplus2g4 commented on src/main/java/erel/task/Deadline.java:
```diff
@@ -0,0 +1,66 @@
+     */
+    @Override
+    public String toFileFormat() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```
Avoid magic string, use a constant

xplus2g4 commented on src/main/java/erel/task/Event.java:
```diff
@@ -0,0 +1,72 @@
+package erel.task;
```
Avoid magic string, use a constant

xplus2g4 commented on src/main/java/erel/task/Task.java:
```diff
@@ -0,0 +1,69 @@
+ * Represents a generic task with a name and a completion status.
+ * This class serves as the base class for specific task types like `Todo`, `Deadline`, and `Event`.
+ */
+public class Task {
```
If task is not going to be created directly, try using an abstract class

xplus2g4 commented on src/main/java/erel/ui/Ui.java:
```diff
@@ -0,0 +1,135 @@
+import java.util.List;
+import java.util.Scanner;
+
+public class Ui {
```
class header

xplus2g4 reviewed (commented):
Generally looks pretty good! Mainly a few cosmetic changes requested.

Commit:
Implement GUI for Level-10

Commit:
Merge branch 'branch-Level-10'
Merge Level-10 with no fast-forward

Commit:
Refactor code

Commit:
Add assertions and improve Javadoc in action classes

Some action classes lacked sufficient validation, which could lead to
unexpected behavior or errors when certain conditions were not met.

Assertions have been added to key methods in the action classes to ensure
that task numbers are valid, parameters are not null, and other critical
conditions hold true during execution.

Additionally, the Javadoc for these methods has been enhanced to improve
clarity.

These improvements are important by preventing invalid states and making it easier to work with
the code in the future.

JohannsenLum committed:
Merge pull request #1 from JohannsenLum/branch-A-Assertions

Add assertions and improve Javadoc in action classes

Commit:
Add JavaDocs for command and exception

Commit:
Add JavaDoc comments for task, gui, exception and Ui classes

Task, gui, exceptions and Ui classes lack detailed documentation for their methods and fields. This leads to challenges in understanding their behavior and usage.

Let's add comprehensive JavaDoc comments for the Task, gui, exceptions and Ui classes, including descriptions for methods, parameters, return values, and exceptions. This ensures better understanding and easier maintenance for developers.

Adding JavaDoc comments is crucial to enhance the readability and maintainability of the code.

JohannsenLum committed:
Merge pull request #2 from JohannsenLum/branch-A-CodeQuality

Add JavaDoc comments for task, gui, exception and Ui classes

Commit:
Add JavaDocs for TaskList

Commit:
Fix CheckStyle violations and adhere to A-CodingStandard

Commit:
Add GitHub Actions CI workflow

Commit:
Update Deadline toString to the correct output format

Commit:
Fix test failures due to platform-specific newline characters

Commit:
Normalize newlines in tests to handle platform differences

Commit:
Add reminder functionality to display upcoming tasks based on type

JohannsenLum committed:
Merge pull request #3 from JohannsenLum/branch-B-Reminders

Add reminder functionality to display upcoming tasks based on type

Commit:
Update avatars for Erel Bot and user

Erel Bot's avatar has been replaced with a friendly, futuristic chatbot
mascot to enhance its visual appeal and reflect its role as an AI
assistant. The user avatar has been updated to a modern gamer-themed
character to better align with the desired identity.

Commit:
Add title to main UI

Commit:
Update Ui.png

Commit:
Update bye command to wait before exit

Commit:
Update ExitAction to pass CheckStyle

JohannsenLum committed:
Update README.md

This update introduces the User Guide for the chatbot, providing users with a comprehensive overview of its features and how to interact with the application. The guide includes detailed instructions for each available command, such as todo, deadline, event, list, mark, unmark, delete, find, and exit.

Additionally, the Table of Contents (TOC) has been added to allow easy navigation to different sections of the user guide. The TOC includes links to sections like Features, Usage, and Command Summary, giving users quick access to key information.

JohannsenLum committed:
Update README.md

Commit:
Update runtest.sh runtest.bat to remove previous data

JohannsenLum committed:
Update README.md

JohannsenLum committed:
Update README.md

JohannsenLum committed:
Update README.md

JohannsenLum committed:
Update README.md

JohannsenLum committed:
Update README.md

Commit:
Refactor Parser to remove magic numbers

Commit:
Merge branch 'master' of https://github.com/JohannsenLum/ip

JohannsenLum committed:
Update README.md with 'remind' feature details

- Add feature extension 'remind' 
- Included instructions on how to use feature 'remind'
- Updated usage example to reflect feature 'remind'

This improves clarity for users and ensures they understand how to utilize the new feature.

Commit:
Enhance GUI by adding main.css

Commit:
Merge branch 'branch-A-BetterGui'

Commit:
Update JavaFX Version in MainWindow.fxml

Commit:
Update JavaFx Version in DialogBox.fxml

Commit:
Update Ui.png to include GUI changes

JohannsenLum committed:
Update README.md

JohannsenLum committed:
Update README.md
